SwaMe.Desktop/Form1.cs

```diff
@@ -0,0 +1,272 @@
+using System.ComponentModel;
+using System.Data;
+using System.Diagnostics;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+
+namespace SwaMe.Desktop
+{
+    public partial class Form1 : Form
+    {
+
+        private string inputFilePath = @"c:\wiffs\collinsb_I180316_005_SW-A.mzML";
+        private string irtFilePath = @"c:\wiffs\hroest_DIA_iRT.TraML";
+        private CancellationTokenSource _cts;
+
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private void Form1_Load(object sender, EventArgs e)
+        {
+
+        }
+
+        private void choose_file_click(object sender, EventArgs e)
+        {
+            using (OpenFileDialog openFileDialog = new OpenFileDialog())
+            {
+                openFileDialog.InitialDirectory = "c:\\";
+                openFileDialog.Filter = "mzml files (*.mzml)|*.mzml|All files (*.*)|*.*";
+                openFileDialog.FilterIndex = 2;
+                openFileDialog.RestoreDirectory = true;
+
+                if (openFileDialog.ShowDialog() == DialogResult.OK)
+                {
+                    //Get the path of specified file
+                    inputFilePath = openFileDialog.FileName;
+                    fileNameLabel.Text = openFileDialog.SafeFileName;
+                }
+            }
+        }
+
+        private void ChooseSpectralLibraryButton_Click(object sender, EventArgs e)
+        {
+            using (OpenFileDialog openFileDialog = new OpenFileDialog())
+            {
+                openFileDialog.InitialDirectory = "c:\\";
+                openFileDialog.Filter = "mzml files (*.mzml)|*.mzml|All files (*.*)|*.*";
+                openFileDialog.FilterIndex = 2;
+                openFileDialog.RestoreDirectory = true;
+
+                if (openFileDialog.ShowDialog() == DialogResult.OK)
+                {
+                    //Get the path of specified file
+
+                    irtFilePath = openFileDialog.FileName;
+                    SpectralLibraryLabel.Text = openFileDialog.SafeFileName;
+                }
+            }
+        }
+
+        private void Form1_Load_1(object sender, EventArgs e)
+        {
+
+        }
+
+        private void label2_Click(object sender, EventArgs e)
+        {
+
+        }
+
+        private void label4_Click(object sender, EventArgs e)
+        {
+
+        }
+
+        private void label10_Click(object sender, EventArgs e)
+        {
+
+        }
+
+        private void maskedTextBox5_MaskInputRejected(object sender, MaskInputRejectedEventArgs e)
+        {
+
+        }
+
+        private void groupBox3_Enter(object sender, EventArgs e)
+        {
+
+
+        }
+
+        private static void CheckFileIsReadableOrComplain(string inputFilePath)
+        {
+            try
+            {
+                Stream stream = new FileStream(inputFilePath, FileMode.Open);
```

Comment:
See my fix elsewhere - add FileAccess.ReadOnly (I think) as a third parameter otherwise this fails on read-only files.